addons/l10n_vn_edi/wizard/l10n_vn_edi_cancellation_request.py

```diff
@@ -0,0 +1,30 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class L10nVNEdiCancellation(models.TransientModel):
+    _name = 'l10n_vn_edi.cancellation'
+    _description = 'E-invoice cancellation wizard'
+
+    invoice_id = fields.Many2one(
+        comodel_name='account.move',
+        string='Invoice to cancel',
+    )
+    reason = fields.Char(
+        string='Reason',
+        required=True,
+    )
+    agreement_document_name = fields.Char(
+        string='Agreement Name',
+    )
+    agreement_document_date = fields.Datetime(
+        string='Agreement Date',
+    )
+
+    def button_request_cancel(self):
```

Comment:
Hm I'll come back to you later on that.
Most likely adding a message in the chatter would be a great idea :smile: But I need to think if it's enough or if we need more

Reply:
Thank you for the feedback! The cancellation flow was really lacking...
It now do things a bit better:

- Cancelling will now post a message with the cancellation reason
- It will now also call button_cancel so that the invoice state is also updated to cancelled (No idea how I missed that)
- Improved the button_draft extend to be correct (only reset fields when going from cancelled to draft, as the comment said)